### PR TITLE
virtual wan/gateway disconnect vpn connections

### PR DESCRIPTION
**Pull request by ninweninwe**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualWan.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualWan.json]:
```suggestion
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/p2svpnGateways/{gatewayName}/disconnectP2sVpnConnections": {
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualWan.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualWan.json]: Please name this `parameters` to keep consistency with other specs

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualWan.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualWan.json]: Example name should be descriptive and human-readable (e.g. "Disconnect P2sVpnConnections from P2sVpnGateway")

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualWan.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualWan.json]: Please name this definition to something like "DisconnectP2sVpnConnectionsParameters". The name `P2SVpnConnectionRequest` sounds like this object is a request for VPN connection

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json]:
```suggestion
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworkGateways/{gatewayName}/disconnectVirtualNetworkGatewayVpnConnections": {
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json]:
```suggestion
        "operationId": "VirtualNetworkGateways_DisconnectVirtualNetworkGatewayVpnConnections",
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json]: Please rename this to `virtualNetworkGatewayName` for the sake of consistency

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json]: Should be name of virtual network gateway

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json]: Please name this `parameters` to keep consistency with other specs

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json]: Example name should be descriptive and human-readable (e.g. "Disconnect P2sVpnConnections from virtual network gateway")

**Review by anton-evseev — CHANGES_REQUESTED** [review]
Please also resolve check errors
cc: @v-shhatt @solhaile

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualNetworkGateway.json]:
```suggestion
            "description": "Request successful. The operation disconnects the requested vpn connections."
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualWan.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/virtualWan.json]:
```suggestion
            "description": "Request successful. The operation disconnects the requested vpn connections."
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/examples/VirtualNetworkGatewaysDisconnectP2sVpnConnections.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/examples/VirtualNetworkGatewaysDisconnectP2sVpnConnections.json]:
```suggestion
      "virtualNetworkGatewayName": "vpngateway",
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/examples/P2sVpnGatewaysDisconnectP2sVpnConnections.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/examples/P2sVpnGatewaysDisconnectP2sVpnConnections.json]:
```suggestion
      "p2sVpnGatewayName": "p2svpngateway",
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/examples/P2sVpnGatewaysDisconnectP2sVpnConnections.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/examples/P2sVpnGatewaysDisconnectP2sVpnConnections.json]: In specs you defined that 200 doesn't return anything, but in example you use empty string. This causes Link checker to throw an error

File: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/examples/VirtualNetworkGatewaysDisconnectP2sVpnConnections.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/examples/VirtualNetworkGatewaysDisconnectP2sVpnConnections.json]: In specs you defined that 200 doesn't return anything, but in example you use empty string. This causes Link checker to throw an error

**Review by anton-evseev — CHANGES_REQUESTED** [review]

**Comment by number213** [comment]
@NullMDR `LintDiff` and `BreakingChange` are failing with
```
Error: (node:5502) UnhandledPromiseRejectionWarning: TypeError: volume[member].bind is not a function
    at patchFilesystem (/home/vsts/work/1/s/node_modules/autorest/dist/static-loader.js:277:43)
    at /home/vsts/work/1/s/node_modules/autorest/dist/static-loader.js:281:18
    at Object.global.staticloader.undo (/home/vsts/work/1/s/node_modules/autorest/dist/static-loader.js:163:7)
    at process.exit (/home/vsts/work/1/s/node_modules/autorest/dist/static-loader.js:170:27)
    at main (/home/vsts/.autorest/@microsoft.azure_autorest-core@2.0.4407/node_modules/@microsoft.azure/autorest-core/dist/app.js:517:21)
(node:5502) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 5)
(node:5502) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
I think this is related to https://github.com/Azure/autorest/issues/3279. Could you please take a look?

**Comment by NullMDR** [comment]
@number123 This bug was recently fixed on latest master, please update your network-december-release branch.

**Comment by ninweninwe** [comment]
> @number123 This bug was recently fixed on latest master, please update your network-december-release branch.

merge with the master of azure rest api specs

**Comment by number213** [comment]
@ninweninwe I've opened separate PR to sync master and release branch: https://github.com/Azure/azure-rest-api-specs/pull/7764
Otherwise yours would be hard to review

**Comment by ninweninwe** [comment]
Do you mean I need to re-sync and do another PR?

**Comment by number213** [comment]
@ninweninwe you can reset your branch to your last commit (https://github.com/Azure/azure-rest-api-specs/pull/7747/commits/e13339c74c5f0ecce2750b020400517d3bf2b02e) or just wait until the PR with sync gets merged

**Comment by ninweninwe** [comment]
> @ninweninwe you can reset your branch to your last commit (e13339c) or just wait until the PR with sync gets merged

reset the commit to (e13339c)

**Comment by number213** [comment]
@ninweninwe thanks, I'll let you know when the other PR gets merged

**Comment by number213** [comment]
Merge PR had a lot of failures because of other RPs, so cherry picked changes to CI into this PR. @ninweninwe there are still failed validation checks, please take a look

**Comment by number213** [comment]
Looks good, but could you please resolve PrettierCheck? You need to run JSON prettifier on your files. You can do that with `npm run-script prettier-fix` (but it will run against all spec files, so please modify package.json so the fix would target only network directory to save time)

**Review by ravbhatnagar — APPROVED** [review]

**Comment by ninweninwe** [comment]
> Looks good, but could you please resolve PrettierCheck? You need to run JSON prettifier on your files. You can do that with npm run-script prettier-fix (but it will run against all spec files, so please modify package.json so the fix would target only network directory to save time)

npm run-script prettier-fix.  But got this error.  Any idea?
[error] Unable to expand glob patterns: specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/*.json !**/node_modules/** !./node_modules/** !**/.{git,svn,hg}/** !./.{git,svn,hg}/**
[error] Cannot find module 'E:\az-github\azure-rest-api-specs\scripts\prettier-swagger-plugin' from 'E:\az-github\azure-rest-api-specs'

**Comment by ninweninwe** [comment]
> Looks good, but could you please resolve PrettierCheck? You need to run JSON prettifier on your files. You can do that with npm run-script prettier-fix (but it will run against all spec files, so please modify package.json so the fix would target only network directory to save time)

is there a way to know where the style issue is?  I could not make prettier working in windows.

**Comment by NullMDR** [comment]
@ninweninwe Please sync(rebase) on latest release branch and follow https://aka.ms/AA6h31t to fix

**Comment by NullMDR** [comment]
@ninweninwe Please run `npm install; npm run prettier -- --write "specification/network/**/*.json"`

**Comment by ninweninwe** [comment]
> @ninweninwe Please run npm install; npm run prettier -- --write "specification/network/**/*.json"

I have prettier installed.  It does not work.

[E:\az-github\azure-rest-api-specs\specification\network\resource-manager\Microsoft.Network\stable\2019-11-01]npm run prettier --write "virtualWan.json"

> azure-rest-api-specs-tests@0.1.0 prettier E:\az-github\azure-rest-api-specs
> prettier "virtualWan.json"

[error] No matching files. Patterns tried: virtualWan.json !**/node_modules/** !./node_modules/** !**/.{git,svn,hg}/** !./.{git,svn,hg}/**
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! azure-rest-api-specs-tests@0.1.0 prettier: `prettier "virtualWan.json"`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the azure-rest-api-specs-tests@0.1.0 prettier script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ninwe\AppData\Roaming\npm-cache\_logs\2019-11-14T17_23_25_482Z-debug.log

**Comment by ninweninwe** [comment]
[E:\az-github\azure-rest-api-specs\specification\network\resource-manager\Microsoft.Network\stable\2019-11-01]npm run prettier -- --write "virtualWan.json"

> azure-rest-api-specs-tests@0.1.0 prettier E:\az-github\azure-rest-api-specs
> prettier "--write" "virtualWan.json"

[error] No matching files. Patterns tried: virtualWan.json !**/node_modules/** !./node_modules/** !**/.{git,svn,hg}/** !./.{git,svn,hg}/**
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! azure-rest-api-specs-tests@0.1.0 prettier: `prettier "--write" "virtualWan.json"`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the azure-rest-api-specs-tests@0.1.0 prettier script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

**Comment by ninweninwe** [comment]
> @ninweninwe Please run npm install; npm run prettier -- --write "specification/network/**/*.json"

actually, I tried other json prettier tool and could not tell any difference of the current files.

**Comment by ninweninwe** [comment]
https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md

I tried the cmd in the master branch.  It did not work for me.

[E:\azure-github\azure-rest-api-specs]npm -v
6.13.0

**Comment by ninweninwe** [comment]
The PR failed on The job running on agent Azure Pipelines 27 ran longer than the maximum time of 180 minutes.  That prettier check can be ignored.   Can anyone trigger another build? @number213

**Comment by NullMDR** [comment]
@ninweninwe You should use `npm run prettier -- --write "specification/network/**/*/virtualWan.json"` even if you are in some sub folder.

**Comment by NullMDR** [comment]
And for the java sdk timeout issue I suggest to remove unused tags in batch section in readme.java.md to speedup generation

**Comment by ninweninwe** [comment]
> @ninweninwe You should use npm run prettier -- --write "specification/network/**/*/virtualWan.json" even if you are in some sub folder.

> azure-rest-api-specs-tests@0.1.0 prettier E:\az-github\azure-rest-api-specs
> prettier "--write" "specification/network/**/*/virtualWan.json"

[error] No matching files. Patterns tried: "--write" "specification/network/**/*/virtualWan.json" !**/node_modules/** !./node_modules/** !**/.{git,svn,hg}/** !./.{git,svn,hg}/**
npm ERR! code ELIFECYCLE
npm ERR! errno 2

still not working

**Comment by ninweninwe** [comment]
> And for the java sdk timeout issue I suggest to remove unused tags in batch section in readme.java.md to speedup generation

can you give me more details?  you mean this?  why does it not have 2019-11 tag?
``` yaml $(java) && $(multiapi)
batch:
  - tag: package-2019-09
```

**Comment by number213** [comment]
The issues with prettier was that you ran it against `virtualWan.json` but the issue was with example files.

I ran the command and pushed prettified files as it's taking too much time to resolve.
```
npm run prettier -- --write "specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/examples/*.json"
```

**Comment by ninweninwe** [comment]
> The issues with prettier was that you ran it against virtualWan.json but the issue was with example files.
> I ran the command and pushed prettified files as it's taking too much time to resolve.
> npm run prettier -- --write "specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/examples/*.json"

[error] No matching files. Patterns tried: "--write" "specification/network/resource-manager/Microsoft.Network/stable/2019-11-01/examples/*.json" !**/node_modules/** !./node_modules/** !**/.{git,svn,hg}/** !./.{git,svn,hg}/**
npm ERR! code ELIFECYCLE
npm ERR! errno 2

I ran the same cmd.  But in my machine, it could not even run.  what else is missing?  do I need to setup anything?  I am using windows and installed prettier already.

**Comment by number213** [comment]
@ninweninwe I ran the command from repo's root (`C:\git\azure-rest-api-specs` for me). Not sure what else might be the reason

**Comment by ninweninwe** [comment]
> @ninweninwe I ran the command from repo's root (C:\git\azure-rest-api-specs for me). Not sure what else might be the reason

I did the same under the repo's root. here is the log.  what is the version of node, npm and prettier on your box?

0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Users\\ninwe\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'prettier',
1 verbose cli   '--',
1 verbose cli   '--write',
1 verbose cli   'specification\\network\\resource-manager\\Microsoft.Network\\stable\\2019-11-01\\examples\\*.json' ]
2 info using npm@6.13.0
3 info using node@v10.15.1
4 verbose run-script [ 'prettier' ]
5 info lifecycle azure-rest-api-specs-tests@0.1.0~prettier: azure-rest-api-specs-tests@0.1.0
6 verbose lifecycle azure-rest-api-specs-tests@0.1.0~prettier: unsafe-perm in lifecycle true
7 verbose lifecycle azure-rest-api-specs-tests@0.1.0~prettier: PATH: C:\Users\ninwe\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;E:\az-github\azure-rest-api-specs\node_modules\.bin;C:\Program Files\ConEmu\ConEmu\Scripts;C:\Program Files\ConEmu;C:\Program Files\ConEmu\ConEmu;C:\Program Files\Common Files\Microsoft Shared\Microsoft Online Services;C:\Program Files (x86)\Common Files\Microsoft Shared\Microsoft Online Services;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files\dotnet\;C:\Program Files\Microsoft\Web Platform Installer\;C:\ProgramData\chocolatey\bin;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;d:\Toolbox\LLVM\bin;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files (x86)\Windows Kits\10\Microsoft Application Virtualization\Sequencer\;C:\Program Files\nodejs\;C:\Program Files\Git\cmd;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\Microsoft Network Monitor 3\;C:\Program Files\PowerShell\6\;C:\Program Files (x86)\dotnet\;C:\Users\ninwe\AppData\Local\CloudBuild;C:\Users\ninwe\AppData\Local\Microsoft\WindowsApps;C:\Users\ninwe\AppData\Local\Dbg\UI;C:\Toolbox\Microsoft VS Code\bin;D:\msys64\usr\bin;;d:\toolbox\Fiddler;C:\Users\ninwe\AppData\Roaming\npm;C:\Users\ninwe\.dotnet\tools;%USERPROFILE%\AppData\Local\Microsoft\WindowsApps
8 verbose lifecycle azure-rest-api-specs-tests@0.1.0~prettier: CWD: E:\az-github\azure-rest-api-specs
9 silly lifecycle azure-rest-api-specs-tests@0.1.0~prettier: Args: [ '-c',
9 silly lifecycle   'prettier "--write" "specification\\network\\resource-manager\\Microsoft.Network\\stable\\2019-11-01\\examples\\*.json"' ]
10 silly lifecycle azure-rest-api-specs-tests@0.1.0~prettier: Returned: code: 2  signal: null
11 info lifecycle azure-rest-api-specs-tests@0.1.0~prettier: Failed to exec prettier script
12 verbose stack Error: azure-rest-api-specs-tests@0.1.0 prettier: `prettier "--write" "specification\network\resource-manager\Microsoft.Network\stable\2019-11-01\examples\*.json"`
12 verbose stack Exit status 2
12 verbose stack     at EventEmitter.<anonymous> (C:\Users\ninwe\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
12 verbose stack     at EventEmitter.emit (events.js:189:13)
12 verbose stack     at ChildProcess.<anonymous> (C:\Users\ninwe\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
12 verbose stack     at ChildProcess.emit (events.js:189:13)
12 verbose stack     at maybeClose (internal/child_process.js:970:16)
12 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
13 verbose pkgid azure-rest-api-specs-tests@0.1.0
14 verbose cwd E:\az-github\azure-rest-api-specs
15 verbose Windows_NT 10.0.18363
16 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\ninwe\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "prettier" "--" "--write" "specification\\network\\resource-manager\\Microsoft.Network\\stable\\2019-11-01\\examples\\*.json"
17 verbose node v10.15.1
18 verbose npm  v6.13.0
19 error code ELIFECYCLE
20 error errno 2
21 error azure-rest-api-specs-tests@0.1.0 prettier: `prettier "--write" "specification\network\resource-manager\Microsoft.Network\stable\2019-11-01\examples\*.json"`
21 error Exit status 2
22 error Failed at the azure-rest-api-specs-tests@0.1.0 prettier script.
22 error This is probably not a problem with npm. There is likely additional logging output above.
23 verbose exit [ 2, true ]

**Review by anton-evseev — APPROVED** [review]

**Comment by number213** [comment]
Good to go, @zikalino please merge

**Pull request merged**
